app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -64,6 +64,15 @@ def in_account_creation_flow?
     user_session[:in_account_creation_flow] || false
   end
 
+  def mfa_selection_attempt_count
+    session[:mfa_attempts] ||= 0
```

Comment:
Should we be using `user_session` so that it gets cleaned up when the user is logged out?